app/controllers/green_lanes/check_your_answers_controller.rb

```diff
@@ -21,34 +21,27 @@ def show
     private
 
     def determine_back_link_path(permitted_params)
-      cat_2_questions_exist = !permitted_params[:ans]['2'].nil?
-      cat_1_questions_dont_exist = permitted_params[:ans]['1'].nil?
+      ans = permitted_params[:ans]
+      category = if ans.nil?
```

Comment:
`dig` simplifies the retrieval of nested values, it seems a good use here:

```
if  ans&.dig('2').present? 
  2
else
  1
end
```

Reply:
ok will try